src/module.ts

```diff
@@ -1,11 +1,61 @@
-import { AppPlugin } from '@grafana/data';
+import { AppPlugin, PluginMeta } from '@grafana/data';
 import { GlobalSettings } from './types';
 import { App } from 'components/App';
 import { ConfigPage } from 'page/ConfigPage';
+import { findSMDataSources } from 'utils';
+import { getBackendSrv } from '@grafana/runtime';
+import { updateSMDatasource } from 'initialization-utils';
+import { getDashboardsNeedingUpdate, importAllDashboards } from 'dashboards/loader';
 
 export const plugin = new AppPlugin<GlobalSettings>().setRootPage(App).addConfigPage({
   title: 'Config',
   icon: 'cog',
   body: ConfigPage,
   id: 'config',
 });
+
+const getPluginSettings = async () =>
+  await getBackendSrv()
+    .fetch<PluginMeta<GlobalSettings>>({
+      url: `/api/plugins/grafana-synthetic-monitoring-app/settings`,
+      method: 'GET',
+      headers: { 'Cache-Control': 'no-store' },
+    })
+    .toPromise()
+    .then((response) => response?.data?.jsonData);
+
+const needToUpdateDSJson = (ds: GlobalSettings, plugin: GlobalSettings): boolean => {
+  if (
+    ds.logs.grafanaName !== plugin.logs.grafanaName ||
+    ds.logs.grafanaName !== plugin.logs.grafanaName ||
+    ds.metrics.grafanaName !== plugin.metrics.grafanaName ||
+    ds.apiHost !== plugin.apiHost
+  ) {
+    return true;
+  }
+  return false;
+};
+
+const preloadInit = async () => {
+  const smDatasources = findSMDataSources();
+  const pluginSettings = await getPluginSettings();
+  // if there is no SM datasource, we have nothing to do. If there is more than 1 we don't know what to do.
```

Comment:
Could maybe short circuit this function earlier based on what comes back in `smDatasources`